File: app/src/main/java/com/shareconnect/database/HistoryItem.java
package com.shareconnect.database;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

@Entity(tableName = "history_items")
@TypeConverters({Converters.class})
public class HistoryItem {
    @PrimaryKey(autoGenerate = true)
    private int id;
    
    private String url;
    private String title;
    private String serviceProvider;
    private String type; // single_video, playlist, channel
    private long timestamp;
    private String profileId;
    private String profileName;
    private boolean sentSuccessfully;
    private String serviceType; // MeTube, Torrent, jDownloader
    
    // Constructors
    public HistoryItem() {}
    
    public HistoryItem(String url, String title, String serviceProvider, String type, 
                      long timestamp, String profileId, String profileName, boolean sentSuccessfully,
                      String serviceType) {
        this.url = url;
        this.title = title;
        this.serviceProvider = serviceProvider;
        this.type = type;
        this.timestamp = timestamp;
        this.profileId = profileId;
        this.profileName = profileName;
        this.sentSuccessfully = sentSuccessfully;
        this.serviceType = serviceType;
    }
    
    // Getters and setters
    public int getId() {
        return id;
    }
    
    public void setId(int id) {
        this.id = id;
    }
    
    public String getUrl() {
        return url;
    }
    
    public void setUrl(String url) {
        this.url = url;
    }
    
    public String getTitle() {
        return title;
    }
    
    public void setTitle(String title) {
        this.title = title;
    }
    
    public String getServiceProvider() {
        return serviceProvider;
    }
    
    public void setServiceProvider(String serviceProvider) {
        this.serviceProvider = serviceProvider;
    }
    
    public String getType() {
        return type;
    }
    
    public void setType(String type) {
        this.type = type;
    }
    
    public long getTimestamp() {
        return timestamp;
    }
    
    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }
    
    public String getProfileId() {
        return profileId;
    }
    
    public void setProfileId(String profileId) {
        this.profileId = profileId;
    }
    
    public String getProfileName() {
        return profileName;
    }
    
    public void setProfileName(String profileName) {
        this.profileName = profileName;
    }
    
    public boolean isSentSuccessfully() {
        return sentSuccessfully;
    }
    
    public void setSentSuccessfully(boolean sentSuccessfully) {
        this.sentSuccessfully = sentSuccessfully;
    }
    
    public String getServiceType() {
        return serviceType;
    }
    
    public void setServiceType(String serviceType) {
        this.serviceType = serviceType;
    }
}

File: Application/src/main/kotlin/com/shareconnect/database/HistoryItem.kt
package com.shareconnect.database

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters

@Entity(tableName = "history_items")
@TypeConverters(Converters::class)
class HistoryItem {
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

    var url: String? = null
    var title: String? = null
    var description: String? = null
    var thumbnailUrl: String? = null
    var serviceProvider: String? = null
    var type: String? = null // single_video, playlist, channel
    var timestamp: Long = 0
    var profileId: String? = null
    var profileName: String? = null
    var isSentSuccessfully: Boolean = false
    var serviceType: String? = null // MeTube, Torrent, jDownloader

    constructor()

    @androidx.room.Ignore
    constructor(
        url: String?,
        title: String?,
        description: String?,
        thumbnailUrl: String?,
        serviceProvider: String?,
        type: String?,
        timestamp: Long,
        profileId: String?,
        profileName: String?,
        isSentSuccessfully: Boolean,
        serviceType: String?
    ) {
        this.url = url
        this.title = title
        this.description = description
        this.thumbnailUrl = thumbnailUrl
        this.serviceProvider = serviceProvider
        this.type = type
        this.timestamp = timestamp
        this.profileId = profileId
        this.profileName = profileName
        this.isSentSuccessfully = isSentSuccessfully
        this.serviceType = serviceType
    }
    
    @androidx.room.Ignore
    constructor(
        url: String?,
        title: String?,
        serviceProvider: String?,
        type: String?,
        timestamp: Long,
        profileId: String?,
        profileName: String?,
        isSentSuccessfully: Boolean,
        serviceType: String?
    ) {
        this.url = url
        this.title = title
        this.serviceProvider = serviceProvider
        this.type = type
        this.timestamp = timestamp
        this.profileId = profileId
        this.profileName = profileName
        this.isSentSuccessfully = isSentSuccessfully
        this.serviceType = serviceType
    }
}